src/classes/Recipe.js

```diff
@@ -3,8 +3,83 @@ class Recipe {
     this.id = recipe.id;
     this.image = recipe.image;
     this.ingredients = recipe.ingredients;
+    this.instructions = recipe.instructions;
+    this.name = recipe.name;
+    this.tags = recipe.tags;
+  }
+  
+  //this is called a private function. We can use it as a helper function
+  //inside of this class - the two underscores are called dunder prototypes
+  __findMatchingIngredient(ingredients, targetIngredient) {
+    return ingredients.find(ingredient => {
+      return ingredient.id === targetIngredient.id;
+    })
+  }
+  
+  calculateCost(ingredientsData) {
+    const totalCost = this.ingredients.reduce((acc, ingredient) => {
+      //use the private findMatchingIng function to match the current
+      //ingredient to the ingredient in the dataset
+      const matchingIngredient = this.__findMatchingIngredient(
+        ingredientsData, 
+        ingredient
+      );
+        // tally up the matching ingredient costs in cents
+      acc += matchingIngredient.estimatedCostInCents;
```

Comment:
where is this .estimatedCostInCents method?

Reply:
It isn't a method, it's a property on the ingredient object.